File: source/livana-be/src/main/java/octguy/livanabe/dto/response/PaymentResponse.java
package octguy.livanabe.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import octguy.livanabe.enums.PaymentMethod;
import octguy.livanabe.enums.PaymentStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Builder
@AllArgsConstructor
public class PaymentResponse {
    
    private UUID id;
    private UUID bookingId;
    private UUID userId;
    private BigDecimal amount;
    private PaymentMethod paymentMethod;
    private PaymentStatus status;
    private String transactionId;
    private String vnpayTransactionNo;
    private String bankCode;
    private String cardType;
    private String orderInfo;
    private LocalDateTime paymentTime;
    private LocalDateTime createdAt;
}

File: source/livana-fe/src/types/response/paymentResponse.ts
export const PaymentStatus = {
  PENDING: "PENDING",
  SUCCESS: "SUCCESS",
  FAILED: "FAILED",
  REFUNDED: "REFUNDED",
} as const;

export type PaymentStatus = (typeof PaymentStatus)[keyof typeof PaymentStatus];

export const PaymentMethod = {
  VNPAY: "VNPAY",
  CASH: "CASH",
} as const;

export type PaymentMethod = (typeof PaymentMethod)[keyof typeof PaymentMethod];

export const BookingType = {
  HOME: "HOME",
  EXPERIENCE: "EXPERIENCE",
} as const;

export type BookingType = (typeof BookingType)[keyof typeof BookingType];

export interface PaymentResponse {
  id: string;
  bookingId: string;
  userId: string;
  amount: number;
  paymentMethod: PaymentMethod;
  status: PaymentStatus;
  transactionId: string;
  vnpayTransactionNo: string | null;
  bankCode: string | null;
  cardType: string | null;
  orderInfo: string;
  paymentTime: string | null;
  createdAt: string;
}

export interface VNPayCreateResponse {
  paymentUrl: string;
  transactionId: string;
}
